### PR TITLE
styled-components refactor

### PR DESCRIPTION
**Pull request by dosyara**
Refactored code base to follow single code style.
Added recommendations to README

I suggest taking one new folder per person and checking your own components.
So
@yasgreen93 `story-collection`
@wajerrr `navigation-menu` + `footer` + `logo`
someone: `article`

Please also have a quick look at components which you wrote and top-level components which are used everywhere. And please, review a README file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+
+```npm run start:dev```
+
+Run tests:
+
+`npm run test`
+
+Update test snapshots:
+
+`npm run test:updateSnapshots`
```
Comment (umbobabo) [inline README.md]: There are no other updates regarding tests, I think we can simplify this:
```
`npm run test:update`
```
or if you want to be prepared for the visual tests:
```
`npm run test:unit:update`
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,69 @@
...
+
+Recommendations for styled components.
+
+#### Naming `styled-[component-name]`
+
+For styled components (components able to receive styles from outside) use naming `styled-[component-name]`
+
+#### Use `styled`
+ 
+Avoid using className on html tags, use `` styled(‘tag’)`your styles` ``.
```
Comment (umbobabo) [inline README.md]: We probably should explain why we took this decision. Would help in future discussions with new devs.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,69 @@
...
+
+`npm run test:updateSnapshots`
+
+### Styled components
+
+Recommendations for styled components.
+
+#### Naming `styled-[component-name]`
+
+For styled components (components able to receive styles from outside) use naming `styled-[component-name]`
```
Comment (umbobabo) [inline README.md]: We should probably communicate the fact that by using <StyledComponent /> that wraps functional component we want to enforce reusability without affecting composability opportunities.
The naming syntax has been decided to enforce the distinction between functional/render and styler component.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,69 @@
...
+const Fancy = styled(Basic)`
+	color: hotpink;
+`
+
+render(<Fancy />);
+```
+
+#### Styles in general
+
+* avoid using cascade in your styles
```
Comment (umbobabo) [inline README.md]: I think is better justify point by point so the readme doesn't act only as a technical documentation but also as a learning documentation for junior devs.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,69 @@
...
+* avoid adding multiple classes on one node, use `cx(baseClass, customClasses)` or composition instead
+* if you need significantly change a look of a component, consider exporting it as a separate component (see StyledSubscribeButton/LargeSubscribeButton)
+* avoid having styles in functional components, wrap your functional-component in styled-component
+
+#### Use folders for rich components
+
+If necessary, use folders for components which have a list of internal components.
```
Comment (umbobabo) [inline README.md]: I think we should always use folder now and explain that in this way tests, snapshots, and styled component will be near the base component

File: src/app/components/article/article-headline.js [hunk 6]
```diff
@@ -34,10 +34,10 @@ const headlineClassName = css`
 `;
 
 const ArticleHeadline = ({ subheadline, children }) => (
-  <h1 className={flyTitleClassName}>
-    {subheadline && <span className={subHeadlineClassName}>{subheadline}</span>}
-    <span className={headlineClassName}>{children}</span>
-  </h1>
+  <StyledArticleHeadline>
```
Comment (umbobabo) [inline src/app/components/article/article-headline.js]: Wow...that is really good. God damn readable 👍

File: src/app/components/article/article-publication-details.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+import text from '../../styles/typography';
+import StyledTime from '../styled-time';
+import StyledLink from '../styled-link';
+import StyledShareBar from '../styled-share-bar';
+
+const Section = ({ printSection, publication }) =>
+  printSection ? (
+    <StyledSectionLink href={printSection.url.canonical}>
+      <StyledIconPrintEdition />
+      {`Print version | ${printSection.headline}`}
```
Comment (umbobabo) [inline src/app/components/article/article-publication-details.js]: What is headline is empty?

File: src/app/components/article/article-publication-details.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+`;
+
+const ArticlePublicationDetails = ({
+  datePublished,
+  byline = '',
+  printSection,
+  publication,
+  commentsUri,
+}) => (
+  <StyledPublicationsContainer>
```
Comment (umbobabo) [inline src/app/components/article/article-publication-details.js]: Guys I love this composition...

File: src/app/components/article/article-headline.js [hunk 9]
```diff
@@ -1,17 +1,17 @@
...
-import { css } from 'emotion';
-import spacings from '../styles/spacings';
-import fontFamily from '../styles/font-family';
-import text from '../styles/typography';
-import color from '../styles/color';
+import styled from 'react-emotion';
+import spacings from '../../styles/spacings';
+import fontFamily from '../../styles/font-family';
+import text from '../../styles/typography';
+import color from '../../styles/color';
```
Comment (yasgreen93) [inline src/app/components/article/article-headline.js]: Could we created an index file for styles? That way we would only need one line of imports.
`import { spacings, fontFamily, text, color } from '../../styles';`

File: src/app/components/article/article-publication-details.js [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+Section.defaultProps = {
+  printSection: null,
+  publication: null,
+};
+
+const StyledIconPrintEdition = styled(IconPrintEdition)`
+  svg {
+    width: 25px;
+    height: 25px;
```
Comment (yasgreen93) [inline src/app/components/article/article-publication-details.js]: Let's add this `25` value to `styles/icon-sizes.js`.

File: src/app/components/article/article-publication-details.js [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    color: ${color.beijing};
+    text-decoration: none;
+    border-bottom: none;
+  }
+`;
+
+const StyledPublicationsContainer = styled('div')`
+  width: 100%;
+  color: ${color.moscow};
+  padding: ${spacings.s} 0;
```
Comment (yasgreen93) [inline src/app/components/article/article-publication-details.js]: 0: `${spacings.none}`

File: src/app/components/navigation-menu/navigation-menu.test.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import renderer from 'react-test-renderer';
 import NavigationMenu from './navigation-menu';
-import * as navigationData from '../../../mockNavData.json';
+import * as navigationData from '../../../../mockNavData.json';
```
Comment (yasgreen93) [inline src/app/components/navigation-menu/navigation-menu.test.js]: We can change this to `import navigationData from '../../../../mockNavData.json';`

File: src/app/components/navigation-menu/styled-accordion-list.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+  border: none;
+  background-color: transparent;
+  font-size: ${typography.sizeStep['1']};
+  font-weight: 500;
+  padding: ${spacings.s} ${spacings.l};
+  color: ${color.kiev};
+`;
+
+const iconStyles = css`
+  bottom: 0;
```
Comment (yasgreen93) [inline src/app/components/navigation-menu/styled-accordion-list.js]: Change the 0s where necessary to `${spacings.none}`

File: src/app/components/navigation-menu/styled-accordion-list.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+`;
+
+const StyledAccordionListItem = ({ item }) => (
+  <AccordionListItem
+    item={item}
+    styles={{
+      StyledSection: sectionStyles,
+      StyledHead: headStyles,
+      StyledIconButton: iconStyles,
+      StyledLink: linkStyles,
```
Comment (yasgreen93) [inline src/app/components/navigation-menu/styled-accordion-list.js]: 🎉 💥 !!!

File: src/app/components/navigation-menu/styled-accordion-list.test.js [hunk 15]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import renderer from 'react-test-renderer';
 import StyledAccordionListItem from './styled-accordion-list';
-import * as navigationData from '../../../mockNavData.json';
+import * as navigationData from '../../../../mockNavData.json';
```
Comment (yasgreen93) [inline src/app/components/navigation-menu/styled-accordion-list.test.js]: `import navigationData from '../../../../mockNavData.json';`

**Pull request merged**